src/lib/ui/widgets/CalendarWidget.svelte

```diff
@@ -1,7 +1,129 @@
 <script lang="ts">
 	import type { CalendarWidget } from '$lib/data/widgets';
+	import { json } from '@sveltejs/kit';
 
 	export let widget: CalendarWidget;
+
+	// Initial values for the current date and time
+	let newDate = new Date();
+	let currentdate = String;
+	let currenttime = String;
+
+	// Function to display the current date and time
+	function displayDateTime() {
+		newDate = new Date();
+		currentdate = `${newDate.getDate()}.${newDate.getMonth() + 1}.${newDate.getFullYear()}`;
+		currenttime = `${newDate.getHours()}:${newDate.getMinutes()} Uhr`;
+	}
+
+	displayDateTime();
+
+	// Reload date and time every minute
+	setInterval(() => {
+		displayDateTime();
+	}, 60000);
+
+	// call the api with the year and loacation
+	const year = newDate.getFullYear();
```

Comment:
Das hier muss noch in die `getHolidays` Funktion verschoben werden. Ansonsten wird `year` nur ein einziges mal berechnet. Wenn das Programm länger als ein Jahr läuft, wird sich `year` also nicht updaten